### ISSUE TITLE
Primary Generated Column not working on oracle db.

### ISSUE THREAD

**Issue by vitorveiga** [typeorm/typeorm#595]
Hello,

I was trying a simple example with a Primary generated column in oracle, but i was getting the following error:

> executing query: declare lastId number; begin INSERT INTO "SAMPLE01_POST" ("TITLE", "TEXT", "LIKESCOUNT") VALUES (:TITLE, :TEXT, :LIKESCOUNT) returning "id" into lastId; dbms_output.enable
> ; dbms_output.put_line(lastId); dbms_output.get_line(:ln, :st); end; -- PARAMETERS: ["hello","Hello how are you?",100,{"dir":3003,"type":2001,"maxSize":32767},{"dir":3003,"type":2002}]
> query failed: declare lastId number; begin INSERT INTO "SAMPLE01_POST" ("TITLE", "TEXT", "LIKESCOUNT") VALUES (:TITLE, :TEXT, :LIKESCOUNT) returning "id" into lastId; dbms_output.enable; d
> bms_output.put_line(lastId); dbms_output.get_line(:ln, :st); end; -- PARAMETERS: ["hello","Hello how are you?",100,{"dir":3003,"type":2001,"maxSize":32767},{"dir":3003,"type":2002}]
> executing query: ROLLBACK
> Cannot save. Error:  Error: ORA-06550: line 1, column 141:
> PL/SQL: ORA-00904: "id": invalid identifier
> ORA-06550: line 1, column 30:
> PL/SQL: SQL Statement ignored

```
import "reflect-metadata";
import {createConnection, ConnectionOptions, Column, Entity, PrimaryGeneratedColumn} from "typeorm";

@Entity("SAMPLE01_POST")
class Post {

    @PrimaryGeneratedColumn()
    ID_POST: number;

    @Column('string')
    TITLE: string;

    @Column('string')
    TEXT: string;

    @Column("number", {
        nullable: false
    })
    LIKESCOUNT: number;
}

const options: ConnectionOptions = {
    type: "oracle",
    host: "<host>",
    username: "<user>",
    password: "<pass>",
    port: 1521,
    sid: "<sid>",
    logging: {
        logQueries: true,
        logFailedQueryError: false,
        logSchemaCreation: true
    },
    autoSchemaSync: false,
    entities: [Post]
};

createConnection(options).then(connection => {

    let post = new Post();
    post.TEXT = "Hello how are you?";
    post.TITLE = "hello";
    post.LIKESCOUNT = 100;

    let postRepository = connection.getRepository(Post);

    postRepository
        .save(post)
        .then(post => console.log("Post has been saved: ", post))
        .catch(error => console.log("Cannot save. Error: ", error));

}, error => console.log("Cannot connect: ", error));

````

I was inspecting the OracleDriver and OracleQueryGenerator implementation, and I think the "returning id" is hardcoded, instead of picking the right Primary Generated Column name. Thats correct?

Best Regards

**Comment by pleerock** [comment]
does it work if you rename `ID_POST` to simply `id` ?

**Comment by vitorveiga** [comment]
Yes... It works. But i can not change the oracle database column names. I have at least 15 tables where the primary key (generated) is not "id".  

A clean solution (in my opinion) is changing OracleQueryRunner.ts on line 212 from

`const sql2 = ``declare lastId number; begin ${insertSql} returning "id" into lastId; dbms_output.enable; dbms_output.put_line(lastId); dbms_output.get_line(:ln, :st); end;``;`

to 

`const sql2 = ``declare lastId number; begin ${insertSql} returning \"${generatedColumn.name}\" into lastId; dbms_output.enable; dbms_output.put_line(lastId); dbms_output.get_line(:ln, :st); end;``;`

What do you think about it? "id" will be changed to generatedColumn.name.

**Comment by pleerock** [comment]
yes made it in `0.1.0-alpha.24`, please check it

**Comment by vitorveiga** [comment]
Many thanks: )

> executing query: declare lastId number; begin INSERT INTO "SAMPLE01_POST" ("TITLE", "TEXT", "LIKESCOUNT") VALUES (:TITLE, :TEXT, :LIKESCOUNT) returning "ID_POST" into l
> astId; dbms_output.enable; dbms_output.put_line(lastId); dbms_output.get_line(:ln, :st); end; -- PARAMETERS: ["hello","Hello how are you?",100,{"dir":3003,"type":2001,"
> maxSize":32767},{"dir":3003,"type":2002}]
> executing query: COMMIT
> Post has been saved:  Post {
>   TEXT: 'Hello how are you?',
>   TITLE: 'hello',
>   LIKESCOUNT: 100,
>   ID_POST: 1 }

**Comment by bmackiewicz** [comment]
which last version typeorm return good lastId in oracle, because i have 0.1.12 and i can't get last insert id ?
When i install 0.1.0-alpha.24 i recive last insert id, changes from 0.1.0-alpha.24 ther are't available in 0.1.12 ?

**Comment by ErannaEru** [comment]
Hi Vitorvegia, can we execute a store procedure using typeorm connection()
I tried but got same error code 06550
query failed:
        BEGIN
            swbappscnc.PRFSP_ACCESS_PRICING_PKG.PRFSP_ACCESS_PRICING(10000142,0,0,32767,0,0);
        END;

error: { [Error: ORA-06550: line 3, column 89:
